### PR TITLE
docs: create `contributing.md`

### PR DESCRIPTION
**Pull request by b-chen00**
Let me know if any changes are needed.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+- Make sure that you are using the latest version.
+- Determine if your bug is really a bug and not an error on your side e.g. using incompatible environment components/versions (Make sure that you have read the [documentation](https://spicetify.app/docs/getting-started/). If you are looking for support, you might want to check [this section](#i-have-a-question)).
+- To see if other users have experienced (and potentially already solved) the same issue you are having, check if there is not already a bug report existing for your bug or error in the [bug tracker](https://github.com/spicetify/spicetify-cli/labels/%F0%9F%90%9B%20bug).
+- Also make sure to search the internet (including Stack Overflow) to see if users outside of the GitHub community have discussed the issue.
+- Collect information about the bug:
+  - Stack trace (Traceback)
+  - OS, Platform and Version (Windows, Linux, macOS, x86, ARM)
+  - Version of the interpreter, compiler, SDK, runtime environment, package manager, depending on what seems relevant.
```
Comment (rxri) [inline CONTRIBUTING.md:45]: same as above

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+  - Use the provided [bug template](https://github.com/spicetify/spicetify-cli/issues/new?assignees=&labels=%F0%9F%90%9B+bug&projects=&template=bug_report.yml).
+- Explain the behavior you would expect and the actual behavior.
+- Please provide as much context as possible and describe the *reproduction steps* that someone else can follow to recreate the issue on their own. This usually includes your code. For good bug reports you should isolate the problem and create a reduced test case.
+- Provide the information you collected in the previous section.
+
+<!-- You might want to create an issue template for bugs and errors that can be used as a guide and that defines the structure of the information to be included. If you do so, reference it here in the description. -->
+
+### Suggesting Enhancements
+
+This section guides you through submitting an enhancement suggestion for Spicetify-cli, **including completely new features and minor improvements to existing functionality**. Following these guidelines will help maintainers and the community to understand your suggestion and find related suggestions.
```
Comment (rxri) [inline CONTRIBUTING.md:63]:
```suggestion
This section guides you through submitting an enhancement suggestion for spicetify, **including completely new features and minor improvements to existing functionality**. Following these guidelines will help maintainers and the community to understand your suggestion and find related suggestions.
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+
+#### How Do I Submit a Good Enhancement Suggestion?
+
+Enhancement suggestions are tracked as [GitHub issues](https://github.com/spicetify/spicetify-cli/issues). Create an enhancement suggestion using the provided [feature request template](https://github.com/spicetify/spicetify-cli/issues/new?assignees=&labels=%E2%9C%A8+feature&projects=&template=feature_request.yml).
+
+- Use a **clear and descriptive title** for the issue to identify the suggestion.
+- Provide a **step-by-step description of the suggested enhancement** in as many details as possible.
+- **Describe the current behavior** and **explain which behavior you expected to see instead** and why. At this point you can also tell which alternatives do not work for you.
+- For GUIs, you may want to **include screenshots and animated GIFs** which help you demonstrate the steps or point out the part which the suggestion is related to. You can use [this tool](https://www.cockos.com/licecap/) to record GIFs on macOS and Windows and [this tool](https://github.com/colinkeenan/silentcast) on Linux.
```
Comment (rxri) [inline CONTRIBUTING.md:79]: On macOS they can use built-in screen recorder https://support.apple.com/en-us/102618,
on Linux they should use ShareX if possible
but in the end, we don't expect them to make a video of the issue

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+      ```
+      cd spicetify-cli
+      go build -o spicetify.exe
+      ```
+   * Linux and MacOS
+      ```
+      cd spicetify-cli
+      go build -o spicetify
+      ```
+3. Execute the executable file generated by `go build` using `spicetify` or `./spicetify`.
```
Comment (rxri) [inline CONTRIBUTING.md:103]:
```suggestion
3. Execute the executable file generated by `go build` using `./spicetify` or `./spicetify.exe`.
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+
+> If you want to ask a question, we assume that you have read the available [Documentation](https://spicetify.app/docs/getting-started/).
+
+Before you ask a question, it is best to search for existing [issues](https://github.com/spicetify/spicetify-cli/issues) that might help you. In case you have found a suitable issue and still need clarification, you can write your question in this issue. It is also advisable to search the internet for answers first.
+
+If you then still feel the need to ask a question and need clarification, we recommend the following:
+
+- Open an [issue](https://github.com/spicetify/spicetify-cli/issues/new).
+- Provide as much context as you can about what you're running into.
+- Provide project and platform versions (go, nodejs, npm, etc), depending on what seems relevant.
```
Comment (rxri) [inline CONTRIBUTING.md:23]: again, we don't need this info. Only spotify and spicetify version most of the time

**Comment by b-chen00** [comment]
Messed up the commit message but fixed with the force pushes

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+    <body>[optional]
+
+*   **type:** feat | fix | docs | chore | revert
+    *   **feat:** A new feature
+    *   **fix:** A bug fix
+    *   **docs:** Documentation only changes
+    *   **chore:** Changes to build process, auxiliary tools, libraries, and other things
+    *   **revert:** A reversion to a previous commit
+*   **scope:** Anything specifying place of the commit change
+    *   Use `*` when affecting more than one scope
```
[the remark this answers is not in the document]
Comment (b-chen00) [reply CONTRIBUTING.md]: removed `*` line

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+#### Before Submitting a Bug Report
+
+A good bug report shouldn't leave others needing to chase you up for more information. Therefore, we ask you to investigate carefully, collect information and describe the issue in detail in your report. Please complete the following steps in advance to help us fix any potential bug as fast as possible.
+
+- Make sure that you are using the latest version.
+- Determine if your bug is really a bug and not an error on your side e.g. using incompatible environment components/versions (Make sure that you have read the [documentation](https://spicetify.app/docs/getting-started/). If you are looking for support, you might want to check [this section](#i-have-a-question)).
+- To see if other users have experienced (and potentially already solved) the same issue you are having, check if there is not already a bug report existing for your bug or error in the [bug tracker](https://github.com/spicetify/spicetify-cli/labels/%F0%9F%90%9B%20bug).
+- Also make sure to search the internet (including Stack Overflow) to see if users outside of the GitHub community have discussed the issue.
+- Collect information about the bug:
+  - Stack trace (Traceback)
```
[the remark this answers is not in the document]
Comment (b-chen00) [reply CONTRIBUTING.md]: removed as well

**Comment by rxri** [comment]
commit message(s) inside pr don't matter as long as the pr title follows conventional commit spec

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+### Reporting Bugs
+
+#### Before Submitting a Bug Report
+
+A good bug report shouldn't leave others needing to chase you up for more information. Therefore, we ask you to investigate carefully, collect information and describe the issue in detail in your report. Please complete the following steps in advance to help us fix any potential bug as fast as possible.
+
+- Make sure that you are using the latest version.
+- Determine if your bug is really a bug and not an error on your side e.g. using incompatible environment components/versions (Make sure that you have read the [documentation](https://spicetify.app/docs/getting-started/). If you are looking for support, you might want to check [this section](#i-have-a-question)).
+- To see if other users have experienced (and potentially already solved) the same issue you are having, check if there is not already a bug report existing for your bug or error in the [bug tracker](https://github.com/spicetify/spicetify-cli/labels/%F0%9F%90%9B%20bug).
+- Also make sure to search the internet (including Stack Overflow) to see if users outside of the GitHub community have discussed the issue.
```
[the remark this answers is not in the document]
Comment (b-chen00) [reply CONTRIBUTING.md]: removed

File: CONTRIBUTING.md (hunk 1 above)
Comment (b-chen00) [reply CONTRIBUTING.md]: removed

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+- Collect information about the bug:
+  - Stack trace (Traceback)
+  - OS, Platform and Version (Windows, Linux, macOS, x86, ARM)
+  - Version of the interpreter, compiler, SDK, runtime environment, package manager, depending on what seems relevant.
+  - Possibly your input and the output
```
[the remark this answers is not in the document]
Comment (b-chen00) [reply CONTRIBUTING.md]: removed

File: CONTRIBUTING.md (hunk 3 above)
Comment (b-chen00) [reply CONTRIBUTING.md]: Added note that video/animated GIF is not expected but also listed the built-in screen recorder and replaced silentcast with ShareX for linux if that route were to be taken

File: CONTRIBUTING.md (hunk 5 above)
Comment (b-chen00) [reply CONTRIBUTING.md]: removed and replace with spotify and spicetify version

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -0,0 +1,119 @@
...
+
+> If you want to ask a question, we assume that you have read the available [Documentation](https://spicetify.app/docs/getting-started/).
+
+Before you ask a question, it is best to search for existing [issues](https://github.com/spicetify/spicetify-cli/issues) that might help you. In case you have found a suitable issue and still need clarification, you can write your question in this issue. It is also advisable to search the internet for answers first.
+
+If you then still feel the need to ask a question and need clarification, we recommend the following:
+
+- Open an [issue](https://github.com/spicetify/spicetify-cli/issues/new).
+- Provide both Spicetify and Spotify version.
+
```
Comment (rxri) [inline CONTRIBUTING.md:23]:
```suggestion
- Open an [issue](https://github.com/spicetify/spicetify-cli/issues/new).
- Provide both Spicetify and Spotify version.
- Explain what the problem is.

```

**Review by rxri — CHANGES_REQUESTED** [review]

**Review by rxri — APPROVED** [review]

**Comment by rxri** [comment]
Ignore the linter

**Review by afonsojramos — APPROVED** [review]
Looks good to me!

**Pull request merged**